azure-pipelines.yml

```diff
@@ -23,6 +23,42 @@ steps:
   clean: true
   fetchDepth: 1
 
+- script: |
+    printfuncs="Print,Printf,Info,Infof,Warning,Warningf,Error,Errorf,Critical,Criticalf,Sprint,Sprintf,Log,Logf,Panic,Panicf,Fatal,Fatalf,Notice,Noticef,Wrap,Wrapf,WithMessage"
+    go vet -all -printfuncs "$printfuncs" ./...
+    exit_code=$?
+    if [[ $? -ne 0 ]]; then
+        printf  "\nThe preceding files contain vetting errors!\n\n"
+        exit ${exit_code}
+    fi
+  displayName: Vet Code
+
+- script: |
+    set -eu -o pipefail
+    files=$(gofmt -l -s . | sort)
+    if [[ ! -z "$files" ]]; then
+     printf "\nThe following files have formatting errors: \n%s\n\n" "${files}"
+     printf "Correct them by executing from the root of the repository:\n"
+     printf "gofmt -l -s -w $(echo ${files} | tr '\n' ' ')\n\n"
+    fi
+  displayName: Lint Code
+
+
+- script: |
```

Comment:
I don't understand why we're running godoc here. The doc will be automatically available when someone hits godoc.org.

What's more important is that we actually document the code. We can help ensure that by using `golint` as it will flag exported symbols that are missing documentation.

Reply:
Removed